testsuite/classlibrary/TestSoundFile.sc

```diff
@@ -53,4 +53,40 @@ TestSoundFile : UnitTest {
 
 	}
 
+	test_cue {
+
+		var cueEvent, testEvent, group;
+		var server = Server(this.class.name);
+		this.bootServer(server);
+
+		// pass in Event
+		testEvent = (foo: true);
+		cueEvent = soundFile.cue(testEvent);
+		this.assert(cueEvent[\foo], "Cue should take an Event as first argument");
+
+		// playNow argument
+		cueEvent = soundFile.cue((), true);
+		0.5.wait;
+		this.assert(cueEvent.isRunning, "Event should play immediately after cueing");
+		cueEvent.stop;
+
+		// Test settable controls
+		group = Group.new(server);
+		testEvent = (bufferSize: 262144, firstFrame: 100, lastFrame: 200, out: 20, server: server, group: group, addAction: 1, amp: 0);
+		cueEvent = soundFile.cue(testEvent);
+		testEvent.keysDo { |key|
+			this.assertEquals(cueEvent[key], testEvent[key], "Control '%' was correclty set".format(key))
+		};
+
+		// closeWhenDone argument
+		soundFile.openRead;
+		soundFile.cue((lastFrame: 10), true, true);
+		0.5.wait;
+		this.assert(soundFile.isOpen.not, "SoundFile should close when finished playing");
+
+		// test cleanup
+		Buffer.freeAll;
+		server.quit.remove;
+
+	}
```

Comment:
this should be split into several tests.; one for `playNow`; one for setting controls; and one for `closeWhenDone`. there should be additional tests for when `playNow` is set to false, and when `closeWhenDone` is set to false.

Reply:
I don't mind separating this into multiple tests. The reason I lumped them together is because they all need the server to be booted. We've already discussed why we don't want to make use of `setUp` and `tearDown` to boot the server in this particular UnitTest in a previous PR (https://github.com/supercollider/supercollider/pull/3712#issuecomment-387266261). Maybe we need to create a separate UnitTest for cue?

Reply:
Yeah, you could move those tests into a separate `TestSoundFile_Cue` or `TestSoundFile_Server`.